calculator.go

```diff
@@ -3,30 +3,99 @@ package calculator
 
 import (
 	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
 )
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-	return a + b
+// Add takes two or subsequent numbers and returns the result of adding them together.
+func Add(a, b float64, nums ...float64) float64 {
+	result := a + b
+	for _, n := range nums {
+		result += n
+	}
+	return result
 }
 
-// Subtract takes two numbers and returns the result of subtracting the second
-// from the first.
-func Subtract(a, b float64) float64 {
-	return a - b
+// Subtract takes two or subsequent numbers and returns the result of subtracting the subsequent
+// number from previous.
+func Subtract(a, b float64, nums ...float64) float64 {
+	result := a - b
+	for _, n := range nums {
+		result -= n
+	}
+	return result
 }
 
-// Multiply takes two numbers and returns the result of multiplication of both
-// the numbers
-func Multiply(a, b float64) float64 {
-	return a * b
+// Multiply takes two or subsequent numbers and returns the result of multiplication of the
+// given numbers
+func Multiply(a, b float64, nums ...float64) float64 {
+	result := a * b
+	for _, n := range nums {
+		result *= n
+	}
+	return result
 }
 
-// Divide takes two numbers and returns the result of dividing the first number
-// by second number. It returns error if the division operation is not valid
-func Divide(a, b float64) (float64, error) {
+// Divide takes two or subsequent numbers and returns the result of dividing the given numbers.
+// It returns error if the division operation is not valid
+func Divide(a, b float64, nums ...float64) (float64, error) {
 	if b == 0 {
 		return 0, errors.New("Divide by Zero")
 	}
-	return a / b, nil
+	result := a / b
+
+	for _, n := range nums {
+		if n == 0 {
+			return 0, errors.New("Divide by Zero")
+		}
+		result /= n
+	}
+	return result, nil
+}
+
+// SquareRoot returns the square root of the provided number, or an error if the input
+// number is invalid
+func SquareRoot(a float64) (float64, error) {
+	if a < 0 {
+		return 0, fmt.Errorf("bad input %f; square root is undefined for negative values", a)
+	}
+	return math.Sqrt(a), nil
+}
+
+// Evaluate processes the input string as two operands separated by the operator
+// and returns the result of that operation, or an error if the string is not in
+// the acceptable form "operand1 operator operand2" with leading and embedded spaces
```

Comment:
What operators are supported? Can you give a few examples of valid inputs?

Reply:
Yes, updated the comment